### PR TITLE
Np 45622 delete publication with nva provided doi

### PR DESCRIPTION
**Pull request by torbjokv**
https://unit.atlassian.net/browse/NP-45622

### CODE REVIEW

**Review by anetteOlli — APPROVED** [review]

File: datacite-registered-doi-handler/src/main/java/no/unit/nva/datacite/handlers/RegisteredDoiEventHandler.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    @Override
+    protected Void processInputPayload(DoiUpdateRequestEvent input,
+                                       AwsEventBridgeEvent<AwsEventBridgeDetail<DoiUpdateRequestEvent>> event,
+                                       Context context) {
+        var doi  = Doi.fromUri(input.getDoi());
+        logger.info(RECEIVED_REQUEST_TO_MAKE_DOI_REGISTERED_LOG, doi.getUri(), input.getPublicationId(),
+                    input.getCustomerId());
+        try {
+            dataCiteMetadataResolver.getDataCiteMetadataXml(input.getPublicationId());
```
Comment (brinxmat) [inline datacite-registered-doi-handler/src/main/java/no/unit/nva/datacite/handlers/RegisteredDoiEventHandler.java:51]: The naming here :( not your fault

File: datacite-registered-doi-handler/src/main/java/no/unit/nva/datacite/handlers/RegisteredDoiEventHandler.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import nva.commons.core.JacocoGenerated;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RegisteredDoiEventHandler extends DestinationsEventBridgeEventHandler<DoiUpdateRequestEvent, Void> {
+    private static final Logger logger = LoggerFactory.getLogger(RegisteredDoiEventHandler.class);
+    private final DoiClient doiClient;
+    private final DataCiteMetadataResolver dataCiteMetadataResolver;
+    public static final String SHOULD_REMOVE_METADATA_LOG_MESSAGE =
+        "Publication with id {} is 410 Gone, assuming it is deleted and doi should have metadata removed";
```
Comment (brinxmat) [inline datacite-registered-doi-handler/src/main/java/no/unit/nva/datacite/handlers/RegisteredDoiEventHandler.java:25]: Not sure if it is better, or more correct, but I prefer not leaking details to logs `Request for publication {} returned 410 Gone. Any Findable DOI associated with this URI will transition to Registered DOI.`

File: datacite-registered-doi-handler/src/main/java/no/unit/nva/datacite/handlers/RegisteredDoiEventHandler.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    private static final String RECEIVED_REQUEST_TO_MAKE_DOI_REGISTERED_LOG =
+        "Received request to unpublish doi (make registered) for DOI {}, publication id {} and customer id {}";
```
Comment (brinxmat) [inline datacite-registered-doi-handler/src/main/java/no/unit/nva/datacite/handlers/RegisteredDoiEventHandler.java:27]: `Will attempt to transition DOI {} to Registered DOI (for publication {} and customer {})` 

Do we get a matching message when this is successful?

**Review by brinxmat — APPROVED** [review]

File: datacite-registered-doi-handler/src/main/java/no/unit/nva/datacite/handlers/RegisteredDoiEventHandler.java (hunk 3 above)
Comment (torbjokv) [reply datacite-registered-doi-handler/src/main/java/no/unit/nva/datacite/handlers/RegisteredDoiEventHandler.java]: yup! SUCCESSFUL_DOI_REGISTERED

File: datacite-registered-doi-handler/src/main/java/no/unit/nva/datacite/handlers/RegisteredDoiEventHandler.java (hunk 2 above)
Comment (torbjokv) [reply datacite-registered-doi-handler/src/main/java/no/unit/nva/datacite/handlers/RegisteredDoiEventHandler.java]: probably make sense to not tie the implementation to the logs here yes, fixed

**Pull request merged**